cql3/functions/time_uuid_fcts.hh

```diff
@@ -111,7 +111,7 @@ make_date_of_fct() {
             return {};
         }
         auto ts = db_clock::time_point(db_clock::duration(UUID_gen::unix_timestamp(UUID_gen::get_UUID(*bb))));
-        return {timestamp_type->decompose(date_type_native_type{ts})};
+        return {timestamp_type->decompose(ts)};
     });
```

Comment:
It's strange, but dateof() returns a timestamp instead of a date. Cassandra does the same, and did so since timestamp type was added.